packages/labs/ssr/src/lib/element-renderer.ts

```diff
@@ -8,16 +8,77 @@
 
 export type Constructor<T> = {new (): T};
 
+import {createRequire} from 'module';
+const require = createRequire(import.meta.url);
+
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const escapeHtml = require('escape-html') as typeof import('escape-html');
 
 import {RenderInfo} from './render-lit-html.js';
 
+export type ElementRendererConstructor = (new (
+  tagName: string
+) => ElementRenderer) &
+  typeof ElementRenderer;
+
+type AttributesMap = Map<string, string>;
+
+export const getElementRenderer = (
```

Comment:
I think we're going to want some kind of renderer registry here so we can store renderers with some kind of precedence info. We could have a registerRenderer method that takes a renderer and a "base" renderer that it supersedes. Or we could offer `registerRenderer(tagName, renderer)` that has the highest precedence.

Reply:
Discussed and agreed to handle in the future.